pipelines/batch/src/main/java/com/google/fhir/analytics/EtlUtils.java

```diff
@@ -46,46 +63,144 @@ static void logMetrics(MetricResults metricResults) {
   }
 
   /**
-   * Runs the given `pipeline` and if the output is to a Parquet DWH, also writes a timestamp file
-   * indicating when the pipeline was started. This is useful for future incremental runs.
+   * Runs the given `pipelines` and if the output is to a Parquet DWH, also writes a timestamp file
+   * indicating when the pipelines were started. This is useful for future incremental runs. The
+   * pipelines are executed by the given {@code executor} pool in parallel.
    *
-   * @return the result from running the pipeline.
+   * @param pipelines the pipelines to be executed
+   * @param options the {@link FhirEtlOptions} to be used by the {@code pipelines}
+   * @param executor the {@link Executor} to be used while executing the {@code pipelines}
+   * @return the result from running the pipelines.
    * @throws IOException if writing the timestamp file fails.
    */
-  static PipelineResult runPipelineWithTimestamp(Pipeline pipeline, FhirEtlOptions options)
-      throws IOException {
+  static List<PipelineResult> runMultiplePipelinesWithTimestamp(
+      List<Pipeline> pipelines, FhirEtlOptions options, Executor executor) throws IOException {
```

Comment:
As discussed, I agree with you that the `Executor` need not be exposed in the non-private methods, since we initialise the pool with a limited number of threads and is not much of a overhead to recreate them each time.